sys/vm_map.c

```diff
@@ -23,18 +24,22 @@ vm_map_t *vm_map_activate(vm_map_t *map) {
   return old;
 }
 
-vm_map_t *get_user_vm_map() { return thread_self()->td_uspace; }
-vm_map_t *get_kernel_vm_map() { return &kspace; }
+vm_map_t *get_user_vm_map() {
```

Comment:
Just add a `.git/hooks/pre-commit` bash script. It will be run in repo root directory, so you can just call `make format`. Note however that hooks are not versioned, so they are **not** shared between repository clones.

I use another approach, as I like to manually review changes done by automatic formatting. I configured my editor to run the formatter each time before I save a file. I don't know how to do that with vi, but - for the record - this is how I did it with emacs:

```elisp
;; Call clang-format before saving C files
(add-hook 'c-mode-hook 
   (lambda () 
      (add-hook 'before-save-hook 'clang-format-buffer nil 'make-it-local)))
```